core/src/test/java/io/aiven/klaw/service/ClusterApiServiceTest.java

```diff
@@ -331,24 +345,88 @@ public void postSchemaSucess() throws KlawException {
     when(restTemplate.postForEntity(Mockito.anyString(), Mockito.any(), eq(ApiResponse.class)))
-        .thenReturn(response);
+            .thenReturn(response);
+
+    ResponseEntity<ApiResponse> result =
+            clusterApiService.postSchema(schemaRequest, envSel, topicName, 1);
+    assertThat(result.getBody().getResult()).isEqualTo(ApiResultStatus.SUCCESS.value);
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"null","false"})
+  @Order(12)
+  public void postSchemaSucessForceRegisterNullOrFalse(String value) throws KlawException {
+    SchemaRequest schemaRequest = new SchemaRequest();
+    schemaRequest.setSchemafull(SCHEMAFULL);
+    schemaRequest.setForceRegister(value.equals("false")?Boolean.valueOf(value) : null);
+    String envSel = "DEV";
+    String topicName = "testtopic";
+
+    ApiResponse apiResponse = ApiResponse.builder().result(ApiResultStatus.SUCCESS.value).build();
+    ResponseEntity<ApiResponse> response = new ResponseEntity<>(apiResponse, HttpStatus.OK);
+
+    when(handleDbRequests.selectEnvDetails(anyString(), anyInt())).thenReturn(this.env);
+    when(manageDatabase.getClusters(any(KafkaClustersType.class), anyInt()))
+            .thenReturn(clustersHashMap);
+    when(clustersHashMap.get(any())).thenReturn(kwClusters);
+    when(kwClusters.getBootstrapServers()).thenReturn(BOOTSRAP_SERVERS);
+    when(kwClusters.getProtocol()).thenReturn(KafkaSupportedProtocol.PLAINTEXT);
+    when(kwClusters.getClusterName()).thenReturn("cluster");
+    when(restTemplate.postForEntity(Mockito.anyString(), Mockito.any(), eq(ApiResponse.class)))
+            .thenReturn(response);
+
+    ResponseEntity<ApiResponse> result =
+            clusterApiService.postSchema(schemaRequest, envSel, topicName, 1);
+    verify(restTemplate,times(1)).postForEntity(anyString(), clusterSchemaRequestCaptor.capture(),eq(ApiResponse.class));
+    assertThat(result.getBody().getResult()).isEqualTo(ApiResultStatus.SUCCESS.value);
+    ClusterSchemaRequest clusterSchemaRequest = clusterSchemaRequestCaptor.getValue().getBody();
+    assertThat(clusterSchemaRequest.isForceRegister()).isFalse();
+    assertThat(clusterSchemaRequest.getFullSchema()).isEqualTo(SCHEMAFULL);
+  }
+
+  @Test
+  @Order(12)
+  public void postSchemaSucessForceRegisterTrue() throws KlawException {
+    SchemaRequest schemaRequest = new SchemaRequest();
+    schemaRequest.setSchemafull(SCHEMAFULL);
+    schemaRequest.setForceRegister(true);
+    String envSel = "DEV";
+    String topicName = "testtopic";
+
+    ApiResponse apiResponse = ApiResponse.builder().result(ApiResultStatus.SUCCESS.value).build();
+    ResponseEntity<ApiResponse> response = new ResponseEntity<>(apiResponse, HttpStatus.OK);
+
+    when(handleDbRequests.selectEnvDetails(anyString(), anyInt())).thenReturn(this.env);
+    when(manageDatabase.getClusters(any(KafkaClustersType.class), anyInt()))
+            .thenReturn(clustersHashMap);
+    when(clustersHashMap.get(any())).thenReturn(kwClusters);
```

Comment:
```suggestion
    when(clustersHashMap.get(anyInt())).thenReturn(kwClusters);
```
since `clustersHashMap` has int keys